### PR TITLE
MultiSegmentAudioPlayer: player capable of playing many file segments

### PR DESCRIPTION
**Pull request by Matt54**
The idea is that developers will have classes that represent their audio clips/segments adopt the protocol "StreamableAudioSegment"

StreamableAudioSegment says that their class will have these properties:
- var audioFile: AVAudioFile { get }
- var playbackStartTime: TimeInterval { get }
- var fileStartTime: TimeInterval { get }
- var fileEndTime: TimeInterval { get }
- var completionHandler: AVAudioNodeCompletionHandler? { get }

Then, any array of objects (named segments in below example) that have adopted the StreamableAudioSegment protocol will be able to be scheduled and played by this new audio player using a simple method call:
audioPlayer.playSegments(audioSegments: segments, referenceTimeStamp: studioTime)

Example video using one player and many segments (sorry - no audio):
https://user-images.githubusercontent.com/15333030/140840514-94d5dad4-9ccc-475a-9815-b73b470328f3.mov

Open to suggestions and future development on this concept :)

### CODE REVIEW

**Comment by Matt54** [comment]
Not sure what's going on with all of these Trailing Whitespace Violation callouts. I ran Format File using SwiftFormat

**Comment by aure** [comment]
This is cool, are there some tests that you could write to ensure that this starts off on the right foot?

**Comment by Matt54** [comment]
@aure sorry for the delay (busy couple of weeks). Tests have been added

File: Sources/AudioKit/Nodes/Playback/MultiSegmentAudioPlayer/MultiSegmentAudioPlayer.swift [hunk 1]
```diff
@@ -0,0 +1,126 @@
+// Copyright AudioKit. All Rights Reserved. Revision History at http://github.com/AudioKit/AudioKit/
+
+import AVFoundation
+
+/// audio player that can schedule many file segments
+public class MultiSegmentAudioPlayer: Node {
+    /// Nodes providing input to this node.
+    public var connections: [Node] { [] }
+
```
[the remark this answers is not in the document]
Comment (071088899) [reply Sources/AudioKit/Nodes/Playback/MultiSegmentAudioPlayer/MultiSegmentAudioPlayer.swift:9]: G

**Pull request merged**